/**:
  ros__parameters:
    use_sim_time: false

    # Global common parameters
    tf_broadcaster_plugin: nav2_util::OdomSmoother
    robot_radius: 0.2
    footprint: "[[0.18, 0.18], [0.18, -0.18], [-0.18, -0.18], [-0.18, 0.18]]"
    transform_tolerance: 0.2
    map_subscribe_transient_local: true

    amcl:
      ros__parameters:
        use_sim_time: false
        alpha1: 0.2
        alpha2: 0.2
        alpha3: 0.2
        alpha4: 0.2
        alpha5: 0.2
        base_frame_id: "base_link"
        beam_skip_distance: 0.5
        beam_skip_error_threshold: 0.9
        beam_skip_threshold: 0.3
        do_beamskip: false
        global_frame_id: "map"
        lambda_short: 0.1
        laser_max_range: 20.0
        laser_min_range: 0.12
        laser_model_type: "likelihood_field"
        max_beams: 60
        max_particles: 2000
        min_particles: 500
        odom_frame_id: "odom"
        pf_err: 0.05
        pf_z: 0.99
        recovery_alpha_fast: 0.0
        recovery_alpha_slow: 0.0
        resample_interval: 1
        robot_model_type: "nav2_amcl::DifferentialMotionModel"
        save_pose_rate: 0.5
        sigma_hit: 0.2
        tf_broadcast: true
        update_min_a: 0.2
        update_min_d: 0.25
        z_hit: 0.5
        z_max: 0.05
        z_rand: 0.5
        z_short: 0.05
        scan_topic: "/scan"

    map_server:
      ros__parameters:
        use_sim_time: false
        yaml_filename: ""

    map_saver:
      ros__parameters:
        use_sim_time: false
        save_map_timeout: 5.0
        free_thresh_default: 0.25
        occupied_thresh_default: 0.65

    planner_server:
      ros__parameters:
        use_sim_time: false
        expected_planner_frequency: 20.0
        planner_plugins: ["GridBased"]
        GridBased:
          plugin: "nav2_navfn_planner/NavfnPlanner"
          tolerance: 0.5
          use_astar: false
          allow_unknown: true

    controller_server:
      ros__parameters:
        use_sim_time: false
        controller_frequency: 20.0
        min_x_velocity_threshold: 0.001
        min_y_velocity_threshold: 0.5
        min_theta_velocity_threshold: 0.001
        failure_tolerance: 0.3
        progress_checker_plugin: "progress_checker"
        goal_checker_plugins: ["goal_checker"]
        controller_plugins: ["FollowPath"]
        progress_checker:
          plugin: "nav2_controller::SimpleProgressChecker"
          required_movement_radius: 0.5
          movement_time_allowance: 10.0
        goal_checker:
          plugin: "nav2_controller::SimpleGoalChecker"
          xy_goal_tolerance: 0.15
          yaw_goal_tolerance: 0.1
          stateful: true
        FollowPath:
          plugin: "dwb_core::DWBLocalPlanner"
          debug_trajectory_details: false
          min_vel_x: 0.0
          min_vel_y: 0.0
          max_vel_x: 0.5
          max_vel_y: 0.0
          max_vel_theta: 1.5
          min_speed_xy: 0.0
          max_speed_xy: 0.5
          min_speed_theta: 0.0
          acc_lim_x: 1.0
          acc_lim_y: 0.0
          acc_lim_theta: 2.0
          decel_lim_x: -1.0
          decel_lim_y: 0.0
          decel_lim_theta: -2.0
          vx_samples: 20
          vy_samples: 1
          vtheta_samples: 40
          sim_time: 1.7
          linear_granularity: 0.05
          angular_granularity: 0.025
          transform_tolerance: 0.2
          xy_goal_tolerance: 0.15
          yaw_goal_tolerance: 0.1
          stateful: true
          trans_stopped_velocity: 0.05
          theta_stopped_velocity: 0.1
          critics: ["ObstacleFootprintCritic", "GoalAlignCritic", "PathAlignCritic", "PathDistCritic", "GoalDistCritic"]
          ObstacleFootprintCritic:
            plugin: "dwb_critics::ObstacleFootprintCritic"
          PathAlignCritic:
            plugin: "dwb_critics::PathAlign"
            scale: 32.0
            forward_point_distance: 0.325
          GoalAlignCritic:
            plugin: "dwb_critics::GoalAlign"
            scale: 24.0
            forward_point_distance: 0.325
          PathDistCritic:
            plugin: "dwb_critics::PathDist"
            scale: 32.0
          GoalDistCritic:
            plugin: "dwb_critics::GoalDist"
            scale: 24.0

    smoother_server:
      ros__parameters:
        use_sim_time: false
        smoother_plugins: ["simple_smoother"]
        simple_smoother:
          plugin: "nav2_smoother::SimpleSmoother"
          tolerance: 1.0e-10
          max_its: 1000
          do_refinement: true

    behavior_server:
      ros__parameters:
        use_sim_time: false
        behavior_plugins: ["spin", "backup", "drive_on_heading", "wait"]
        spin:
          plugin: "nav2_behaviors::Spin"
        backup:
          plugin: "nav2_behaviors::BackUp"
        drive_on_heading:
          plugin: "nav2_behaviors::DriveOnHeading"
        wait:
          plugin: "nav2_behaviors::Wait"
        global_frame: map
        robot_base_frame: base_link
        transform_tolerance: 0.2
        simulate_ahead_time: 2.0
        max_rotational_vel: 1.0
        min_rotational_vel: 0.4
        rotational_acc_lim: 3.2

    bt_navigator:
      ros__parameters:
        use_sim_time: false
        global_frame: map
        robot_base_frame: base_link
        transform_tolerance: 0.2
        default_nav_to_pose_bt_xml: "$(find-pkg-share nav2_bt_navigator)/behavior_trees/navigate_w_replanning_time.xml"
        default_nav_through_poses_bt_xml: "$(find-pkg-share nav2_bt_navigator)/behavior_trees/navigate_through_poses_w_replanning_time.xml"
        plugin_lib_names:
        - nav2_compute_path_to_pose_action_bt_node
        - nav2_compute_path_through_poses_action_bt_node
        - nav2_follow_path_action_bt_node
        - nav2_back_up_action_bt_node
        - nav2_spin_action_bt_node
        - nav2_wait_action_bt_node
        - nav2_clear_costmap_service_bt_node
        - nav2_is_stuck_condition_bt_node
        - nav2_goal_reached_condition_bt_node
        - nav2_goal_updated_condition_bt_node
        - nav2_initial_pose_received_condition_bt_node
        - nav2_reinitialize_global_localization_service_bt_node
        - nav2_rate_controller_bt_node
        - nav2_distance_controller_bt_node
        - nav2_speed_controller_bt_node
        - nav2_truncate_path_action_bt_node
        - nav2_undo_path_truncation_action_bt_node
        - nav2_follow_path_goal_checker_bt_node
        - nav2_navigate_through_poses_action_bt_node
        - nav2_navigate_to_pose_action_bt_node
        - nav2_remove_passed_goals_action_bt_node
        - nav2_planner_selector_bt_node
        - nav2_controller_selector_bt_node
        - nav2_goal_checker_selector_bt_node
        - nav2_controller_cancel_bt_node
        - nav2_path_longer_on_approach_bt_node
        - nav2_wait_cancel_bt_node
        - nav2_spin_cancel_bt_node
        - nav2_back_up_cancel_bt_node
        - nav2_drive_on_heading_cancel_bt_node
        - nav2_is_battery_low_condition_bt_node

    waypoint_follower:
      ros__parameters:
        use_sim_time: false
        loop_rate: 20
        stop_on_failure: false
        waypoint_task_executor_plugin: "wait_at_waypoint"
        wait_at_waypoint:
          plugin: "nav2_waypoint_follower::WaitAtWaypoint"
          enabled: true
          waypoint_pause_duration: 2000

    velocity_smoother:
      ros__parameters:
        use_sim_time: false
        smoothing_frequency: 20.0
        scale_velocities: true
        feedback: "OPEN_LOOP"
        max_velocity: [0.5, 0.0, 1.5]
        min_velocity: [-0.5, 0.0, -1.5]
        max_accel: [1.0, 0.0, 2.0]
        max_decel: [-1.0, 0.0, -2.0]
        odom_frame: odom
        base_frame: base_link
        odom_topic: "/odom"
        accel_timeout: 1.0
        decel_timeout: 1.0
        use_stamped_vel: false

    local_costmap:
      local_costmap:
        ros__parameters:
          use_sim_time: false
          plugin_names: ["obstacle_layer", "inflation_layer"]
          plugin_types: ["nav2_costmap_2d::ObstacleLayer", "nav2_costmap_2d::InflationLayer"]
          global_frame: odom
          robot_base_frame: base_link
          update_frequency: 10.0
          publish_frequency: 10.0
          transform_tolerance: 0.2
          resolution: 0.05
          rolling_window: true
          width: 6.0
          height: 6.0
          origin_x: 0.0
          origin_y: 0.0
          inflation_layer:
            cost_scaling_factor: 3.0
            inflation_radius: 0.55
          obstacle_layer:
            max_obstacle_height: 2.0
            observation_sources: scan
            scan:
              topic: /scan
              max_obstacle_height: 1.0
              clearing: true
              marking: true
              data_type: LaserScan
              raytrace_max_range: 12.0
              obstacle_max_range: 10.0
          always_send_full_costmap: true

    global_costmap:
      global_costmap:
        ros__parameters:
          use_sim_time: false
          plugin_names: ["static_layer", "obstacle_layer", "inflation_layer"]
          plugin_types: ["nav2_costmap_2d::StaticLayer", "nav2_costmap_2d::ObstacleLayer", "nav2_costmap_2d::InflationLayer"]
          global_frame: map
          robot_base_frame: base_link
          update_frequency: 5.0
          publish_frequency: 5.0
          transform_tolerance: 0.5
          resolution: 0.05
          track_unknown_space: true
          use_maximum: false
          rolling_window: false
          inflation_layer:
            cost_scaling_factor: 3.5
            inflation_radius: 0.75
          obstacle_layer:
            enabled: true
            max_obstacle_height: 2.0
            observation_sources: scan
            scan:
              topic: /scan
              max_obstacle_height: 1.0
              clearing: true
              marking: true
              data_type: LaserScan
              raytrace_max_range: 12.0
              obstacle_max_range: 10.0
          static_layer:
            map_subscribe_transient_local: true
          always_send_full_costmap: true

    lifecycle_manager:
      ros__parameters:
        use_sim_time: false
        autostart: true
        node_names:
        - controller_server
        - planner_server
        - smoother_server
        - behavior_server
        - bt_navigator
        - waypoint_follower
        - velocity_smoother
        - map_server
        - amcl
        - map_saver
